### ISSUE TITLE
Add wmllint to Travis builds

### ISSUE THREAD

**Issue by Wedge009** [wesnoth/wesnoth#7082]
### Describe the desired feature

`wmllint` errors, genuine or otherwise, have been discussed on and off, most recently in #5799. Commentary in #4381 suggests adding it to the Travis build.

**Comment by Pentarctagon** [comment]
We use Github Actions these days, not Travis :P

**Comment by Wedge009** [comment]
I have no idea what the terminology is, I was just following what was written.

**Comment by sevu** [comment]
Exists for add-ons:
https://github.com/czyzby/wesnoth-wml-linter

It is a bit different though, in that it runs wmllint on mainline (to get the unit ids etc) and the commit/pr in the add-on repo, and filters the output which belongs to mainline.

**Comment by Pentarctagon** [comment]
> I have no idea what the terminology is, I was just following what was written.

For future reference then - travis and github actions are CI service providers. Travis is what we used before switching to github actions.

**Comment by czyzby** [comment]
@Pentarctagon What is the expected behavior? Should the jobs fail if wmllint/wmlindent/wmlscope report any issues? Because right now there are quite a lot warnings if you run those.

**Comment by Pentarctagon** [comment]
Yes, the expectation would be that it should fail. And yeah, it's known there's a bunch of things it warns on. The issue really is that nobody has gone through all those warnings and either fixed the issue or removed the warning for being bad/irrelevant.

**Comment by czyzby** [comment]
What about spellchecking? I imagine it might increase the number of errors significantly, since you have to register all non-dictionary words with `wmllint: general spelling(s)`.

**Comment by Pentarctagon** [comment]
What's the question about spellchecking?

**Comment by czyzby** [comment]
Should it also fail the pipeline or be skipped in the automatic check?

**Comment by Pentarctagon** [comment]
It should fail the pipeline.

**Comment by czyzby** [comment]
OK, last question - could you give me a list of all tools that you want and any command line flags that you'd like to include? I'm assuming we'd like to run `wmllint`, `wmlindent` and `wmlscope` in dry-run mode wherever possible. Any extra flags for `wmlscope`?

**Comment by Pentarctagon** [comment]
wmlindent is already run. I'm not familiar with wmlscope.

**Comment by cooljeanius** [comment]
> wmlindent is already run. I'm not familiar with wmlscope.

I usually end up skipping that one, just because it prints *so much* output that I don't know where to start... might be worth asking @Elvish-Hunter about...

**Comment by CelticMinstrel** [comment]
`wmlscope` is the tool that generates the macro documentation page, maybe that's relevant somehow.

**Comment by cooljeanius** [comment]
> `wmlscope` is the tool that generates the macro documentation page, maybe that's relevant somehow.

Yes, but that's not the mode in which the proposed workflow runs it, though.

**Comment by Pentarctagon** [comment]
Closed with #11309
